feat(functions): Mode aware functions (#415)
mrjones2014/legendary.nvim: doc/table_structures/FUNCTIONS.md

}
```

## Specifying mode

By default, the funciton is shown and run in `*` (all) modes.

> You can use `mode` property to narrow function's scope, so it always run in specified mode:

Reviewer: ```suggestion
You can use `mode` property to narrow function's scope, so it only appears in the specified mode(s):
```
